### ISSUE TITLE
could pvactools be used for  prediction in mice ?

### ISSUE THREAD

**Issue by zhangyafeng1** [griffithlab/pVACtools#554]
* pvactools version: 1.5.4
my pvactools version is 1.5.4 ,  and  I found in the article that NetMHCpan 4.0  predict peptides to any MHC molecule  。 [https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6714459/](url)
 In pvactools version: 1.5.4 ， NetMHCpan version is 4.0 .  
So , could I use pvactools to predict in mice?

![image](https://user-images.githubusercontent.com/52239895/81567944-a4c35700-93cf-11ea-88be-4b893e78c6bf.png)

**Comment by susannasiebert** [comment]
Yes, you can use pVACtools to make predictions for mice but the available alleles are very limited:
```
$ pvacseq valid_alleles | grep H2
H2-IAb
H2-IAd
H2-IEd
$ pvacseq valid_alleles | grep H-2
H-2-Db
H-2-Dd
H-2-Kb
H-2-Kd
H-2-Kk
H-2-Ld
```

**Comment by malachig** [comment]
Fortunately, if you are working with common lab mice (as most people are) there is a good chance it will involve one of these alleles. e.g.

C57BL/6 is: H-2-Kb, H-2-Db, H2-IAb
129 is: H-2-Kb, H-2-Db, H2-IAb
BALB/Cj is: H-2-Kd, H-2-Dd, H-2-Ld, H2-IAd, H2-IEd

http://tools.thermofisher.com/content/sfs/brochures/Mouse_Haplotype_Table.pdf
http://jackson.jax.org/rs/444-BUH-304/images/MHC_h2_haplotypes.pdf

**Comment by zhangyafeng1** [comment]
Tkank you very much!
but when I tested it, I met with some errors.
My command line are as follows

`pvacseq run \
 MC38_somatic_final.vcf.gz \
 MC38_Met_Lines \
 H-2-Kb,H-2-Db \
 NetMHCpan \
 /out_dir/snv_pvac \
 -e 8 \
 -t 2 \
 -p MC38_Met_Lines_phased_vep.vcf.gz \
 --iedb-install-directory  database_dir/mhc_database \
 --normal-sample-name MC38_N_Lines
`


when it running **Top Score Filter** , the following error occurred ,   I didn't met this problem when I used a species human.

`Running Transcript Support Level Filter
Complete
Running Top Score Filter
usage: pvacseq [-h]
               {run,binding_filter,coverage_filter,transcript_support_level_filter,top_score_filter,generate_protein_fasta,generate_condensed_ranked_report,download_example_data,install_vep_plugin,valid_alleles,allele_specific_cutoffs}
               ...

positional arguments:
  {run,binding_filter,coverage_filter,transcript_support_level_filter,top_score_filter,generate_protein_fasta,generate_condensed_ranked_report,download_example_data,install_vep_plugin,valid_alleles,allele_specific_cutoffs}
    run                 Runs the pVACseq pipeline
    binding_filter      Filters variants processed by IEDB by binding score
    coverage_filter     Filters variants processed by IEDB by coverage, vaf,
                        and gene expression
    transcript_support_level_filter
                        Filters variants processed by IEDB by transcript
                        support level
    top_score_filter    Pick the best neoepitope for each variant
    generate_protein_fasta
                        Generate an annotated fasta file from a VCF with
                        protein sequences of mutations and matching wildtypes
    generate_condensed_ranked_report
                        Generate a condensed, ranked report from a pVACseq
                        .all_epitopes.tsv or .filtered.tsv report file.
    download_example_data
                        Downloads example input and output files
    install_vep_plugin  Installs the Wildtype VEP plugin into your VEP_plugins
                        directory
    valid_alleles       Shows a list of valid allele names
    allele_specific_cutoffs
                        Show the allele specific cutoffs

optional arguments:
  -h, --help            show this help message and exit
Error: No command specified
Traceback (most recent call last):
  File "/ifs/TJPROJ3/HWEU/PUBLIC/software/reseq/pvac_python3/anaconda3/bin/pvacseq", line 8, in <module>
    sys.exit(main())
  File "/ifs/TJPROJ3/HWEU/USER/zhangyafeng/sorftware/anaconda3/lib/python3.7/site-packages/tools/pvacseq/main.py", line 92, in main
    args[0].func.main(args[1])
  File "/ifs/TJPROJ3/HWEU/USER/zhangyafeng/sorftware/anaconda3/lib/python3.7/site-packages/tools/pvacseq/run.py", line 164, in main
    pipeline.execute()
  File "/ifs/TJPROJ3/HWEU/USER/zhangyafeng/sorftware/anaconda3/lib/python3.7/site-packages/lib/pipeline.py", line 469, in execute
    PostProcessor(**post_processing_params).execute()
  File "/ifs/TJPROJ3/HWEU/USER/zhangyafeng/sorftware/anaconda3/lib/python3.7/site-packages/lib/post_processor.py", line 33, in execute
    self.execute_top_score_filter()
  File "/ifs/TJPROJ3/HWEU/USER/zhangyafeng/sorftware/anaconda3/lib/python3.7/site-packages/lib/post_processor.py", line 102, in execute_top_score_filter
    TopScoreFilter(self.transcript_support_level_filter_fh.name, self.top_score_filter_fh.name, self.top_score_metric, self.file_type).execute()
  File "/ifs/TJPROJ3/HWEU/USER/zhangyafeng/sorftware/anaconda3/lib/python3.7/site-packages/lib/top_score_filter.py", line 82, in execute
    lowest_transcript_id = re.compile('ENST(\d+)').match(line_with_lowest_transcript_id['Transcript']).group(1)
AttributeError: 'NoneType' object has no attribute 'group'
`

could you give me some help? Thank you!

**Comment by zhangyafeng1** [comment]
I carried out this step alone for testing.

`pvacseq    top_score_filter   MC38_Met_Lines.all_epitopes.tsv   out.tsv`

When I modified the beginning of transcript in input file from ENSMUST to ENST .  The normal execution of the script is over.

I have a little confused . Did I miss some necessary parameters?

**Comment by susannasiebert** [comment]
This looks like a bug. I will look into it further.

**Comment by susannasiebert** [comment]
This bug has been fixed in version 1.5.9.

**Comment by Xiao-Zhong** [comment]
> Yes, you can use pVACtools to make predictions for mice but the available alleles are very limited:
> 
> ```
> $ pvacseq valid_alleles | grep H2
> H2-IAb
> H2-IAd
> H2-IEd
> $ pvacseq valid_alleles | grep H-2
> H-2-Db
> H-2-Dd
> H-2-Kb
> H-2-Kd
> H-2-Kk
> H-2-Ld
> ```

Hi, no thing shows up with the command. 
pvacseq valid_alleles did output alleles but doesn't have H2 or H-2. See attached file.
pvactools version I'm using is Version: 5.2.0.

Could you help on this? Thank you!
